### PR TITLE
Midround Escape Pod Launch

### PR DESCRIPTION
**Pull request by ghost**
Adds computers for the escape pods, which when emagged, allow you to launch the escape pod early.

Will also launch early during code red or code delta.

Adds random shuttle destinations, which scramble their location roundstart.

The pod computers and destinations are not actually on the map yet. They're intended for lavaland, but they'd be usable with the derelict or mine or something if someone wanted to use them.

### CODE REVIEW

**Comment by WJohn** [comment]
:+1: Very neat.

**Comment by KorPhaeron** [comment]
Whoops. Teach me to use the online editor.

File: code/modules/security levels/security levels.dm [hunk 1]
```diff
@@ -26,6 +26,8 @@
 				for(var/obj/machinery/firealarm/FA in world)
 					if(FA.z == ZLEVEL_STATION)
 						FA.update_icon()
+				for(var/obj/machinery/computer/shuttle/pod/pod in world)
```
Comment (RemieRichards) [inline code/modules/security levels/security levels.dm]: NO ```in world```

Comment (KorPhaeron) [inline code/modules/security levels/security levels.dm]: Is there a machinery list or something that's better?

Comment (RemieRichards) [inline code/modules/security levels/security levels.dm]: it's a computer, it might be in a computer list
it's a machine, it might be in the machines list
perhaps even define a pod_computer list, that contains only this type

**Comment by KorPhaeron** [comment]
Turns out it was machines. Fixed the in world for the alarms as well.

**Comment by Jalleo** [comment]
:+1: Is the launch automatic or is it delayed by a few moments? If its the latter could be useful for trapping people then launching it as a way to remove undesirables.

**Comment by KorPhaeron** [comment]
It's automatic, a delay would be possible though.

**Comment by Jalleo** [comment]
A delay overall might be better to give it a feel that the escape pod is not ready compared to a normal escape shuttle call.

**Comment by Anonus** [comment]
A ten second delay would be perfect, yes.

**Comment by MrStonedOne** [comment]
koooooor

koooooooooooor

You touched a file with relative pathing kor.

You know what that means don't you?

You must fix the relative pathing.

**Comment by RemieRichards** [comment]
^

**Comment by KorPhaeron** [comment]
Mergebegging 

@Miauw62 @RemieRichards @Cheridan @phil235

File: code/controllers/subsystem/shuttles/emergency.dm [hunk 2]
```diff
@@ -148,3 +146,27 @@
 		if(.)	return .
 		return ..()
 */
+
+/obj/machinery/computer/shuttle/pod
```
Comment (phil235) [inline code/controllers/subsystem/shuttles/emergency.dm]: why is this here instead of in the same file as obj/machinery/computer/shuttle?

File: code/modules/security levels/security levels.dm [hunk 3]
```diff
@@ -47,15 +51,19 @@
...
+					pod.admin_controlled = 1
 			if(SEC_LEVEL_DELTA)
 				minor_announce(config.alert_desc_delta, "Attention! Delta security level reached!",1)
 				security_level = SEC_LEVEL_DELTA
-				for(var/obj/machinery/firealarm/FA in world)
+				for(var/obj/machinery/firealarm/FA in machines)
 					if(FA.z == ZLEVEL_STATION)
 						FA.update_icon()
+				for(var/obj/machinery/computer/shuttle/pod/pod in machines)
+					pod.admin_controlled = 1
```
Comment (phil235) [inline code/modules/security levels/security levels.dm]: I don't understand what's the point of toggling admin_controlled to 1 at every single security level?

File: code/controllers/subsystem/shuttles/emergency.dm (hunk 2 above)
Comment (KorPhaeron) [inline code/controllers/subsystem/shuttles/emergency.dm]: Because the rest of the emergency shuttle code is there, and all the special shuttle computers (such as the gulag) are in their own files anyway
